src/lifecycle.js

```diff
@@ -1,16 +1,18 @@
 export default Base =>
   class extends Base {
-    UNSAFE_componentWillMount () {
+    constructor(props) {
+      super(props);
+
       this.setStateWithData(this.getDataModel(this.getResolvedState(), true))
```

Comment:
This doesn't work for me. columns is undefined in getDataModel. If I change `super()` in index.js to `super(props)`, it fails when trying to call setState in the constructor.

Reply:
🤔 could you post the steps to reproduce this. 

I didn't realize we are calling `setState` here - which is definitely not right. I can add a function to calculate proper `state` object and use it here. (and make `this. setStateWithData` to use that to get the state)

Reply:
https://github.com/tannerlinsley/react-table/pull/1621